argh_derive/src/lib.rs

```diff
@@ -471,7 +467,11 @@ fn impl_from_args_struct_redact_arg_values<'a>(
             #missing_requirements_ident.err_on_any()?;
 
             let mut __redacted = vec![
-                #cmd_name,
+                if let Some(cmd_name) = __cmd_name.last() {
```

Comment:
This change looks non-trival. 

Not a blocker, but I'm curious why?

Seems like a lot of extra effort, hard to really understand what it's doing, and I'm not sure what the real benefit is of generating an Err(EarlyExit()) instead of what it was doing before.

Reply:
In this case, we'd panic if someone had:

```
#[derive(FromArgs)]
#[argh(subcommand)]
struct Foo { ... }
Foo::redact_arg_values(&[], &[...]); // panics
```

This changes the code to instead return an error, because the user needs to be passing in the subcommand name in order to generate the help string.

The other option we could just not prepend the cmd name to the redacted list, but since this is part of the semantics of subcommands, I figure we should error in that case.

Reply:
Got it! Thanks!